internal/util/observability/observability.go

```diff
@@ -22,31 +22,61 @@ import (
 	"time"
 
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
 	otelsdkresource "go.opentelemetry.io/otel/sdk/resource"
 	otelsdktrace "go.opentelemetry.io/otel/sdk/trace"
 	otelsemconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-
-	"github.com/FerretDB/FerretDB/internal/util/must"
+	"go.uber.org/zap"
 )
 
-// SetupOtel sets up OpenTelemetry exporter with fixed values.
-func SetupOtel(service string) func(context.Context) error {
-	exporter := must.NotFail(otlptracehttp.New(
+// OtelConfig is the configuration for OpenTelemetry.
+type OtelConfig struct {
+	Service  string
+	Version  string
+	Endpoint string
+}
+
+// RunOtel runs the OpenTelemetry system with the given configuration.
+func RunOtel(ctx context.Context, config OtelConfig, logger *zap.SugaredLogger) {
```

Comment:
As discussed, we should either return an error from that function or make a separate construction function that could return an error. FerretDB should shutdown if there was an initialization error.